src/ui/components/trinary_input_string.c

```diff
@@ -235,6 +235,31 @@ static void _render(component_t* component)
     }
 }
 
+// if the current input uniquely identifies a word from the wordlist by prefix, we autocomplete the
+// word.
+static void _maybe_autocomplete(component_t* trinary_input_string)
+{
+    data_t* data = (data_t*)trinary_input_string->data;
+    if (data->wordlist == NULL) {
+        return;
+    }
+    // initial value means no word was found yet.
+    size_t found_word_idx = data->wordlist_size;
+    for (size_t word_idx = 0; word_idx < data->wordlist_size; word_idx++) {
```

Comment:
I think there are ways to optimize this but probably not worth it today.

Reply:
today or maybe ever? it's a small word list, even the whole prefix search to create the keyboard after each tap is not noticeable. or did you mean something else?

Reply:
yeah I actually meant "maybe ever" :)
didn't notice any issues visually.